monitoring/modules.go

```diff
@@ -0,0 +1,209 @@
+/*
+Copyright 2017 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package monitoring
+
+import (
+	"bufio"
+	"context"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/gravitational/satellite/agent/health"
+	pb "github.com/gravitational/satellite/agent/proto/agentpb"
+
+	"github.com/gravitational/trace"
+	log "github.com/sirupsen/logrus"
+)
+
+// NewModuleChecker creates a new kernel module checker
+func NewModuleChecker(names ...string) ModuleChecker {
+	return ModuleChecker{
+		Modules:      names,
+		moduleGetter: moduleGetterFunc(LoadModules),
+	}
+}
+
+// Name returns name of the checker
+func (r ModuleChecker) Name() string {
+	return "module-check"
+}
+
+// Check determines if the modules specified with r.Modules have been loaded
+func (r ModuleChecker) Check(ctx context.Context, reporter health.Reporter) {
+	err := r.check(ctx, reporter)
+	if err != nil {
+		reporter.Add(NewProbeFromErr(r.Name(), "", trace.Wrap(err)))
+		return
+	}
+	reporter.Add(&pb.Probe{Checker: r.Name(), Status: pb.Probe_Running})
+}
+
+func (r ModuleChecker) check(ctx context.Context, reporter health.Reporter) error {
+	modules, err := r.moduleGetter.Modules()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	var errors []error
+	for _, module := range r.Modules {
+		if !modules.HasLoaded(module) {
+			errors = append(errors, trace.NotFound("module %q not loaded", module))
+		}
+	}
+
+	if len(errors) == 1 {
```

Comment:
this logic is already present in `trace.NewAggregate`

Reply:
Unfortunately not but I can fix it in trace